apps/site/assets/ts/hooks/__tests__/useChannelTest.tsx

```diff
@@ -89,4 +96,39 @@ describe("useChannel", () => {
       initialState.things
     ); // still has original things
   });
+
+  test("when passed null topic, channel is not joined", () => {
+    expect(window.channels["mock-channel"]).toBeUndefined();
+    renderHook(() =>
+      useChannel<MockData, MockReducer>(null, mockReducer, initialState)
+    );
+    expect(mockChannel!.join).not.toHaveBeenCalled();
+  });
+
+  test("when passed null topic then string topic, the topic is joined", () => {
+    expect(window.channels["mock-channel"]).toBeUndefined();
+    const { rerender } = renderHook<string | null, MockState>(
+      channelId =>
+        useChannel<MockData, MockReducer>(channelId, mockReducer, initialState),
+      { initialProps: null }
+    );
+    expect(mockChannel!.join).not.toHaveBeenCalled();
+    rerender("mock-channel");
+    expect(mockChannel!.join).toHaveBeenCalled();
+    expect(window.channels["mock-channel"]).toBeDefined();
+  });
+
+  test("when passed string topic then null topic, the topic is left", () => {
```

Comment:
Do we always want to leave the channel on a null?

Reply:
I'd say yes within the context of a single `useChannel` call! If useChannel is passed "A" as a topic and then "B", I'd expect that this would leave channel "A" and join channel "B". In this case, channel B is null so we don't have to join another channel, but we'd still want to leave A.